inference/server/main.py

```diff
@@ -106,6 +110,70 @@ def maybe_add_debug_api_keys():
         raise
 
 
+@app.get("/auth/login/discord")
+async def login_discord(request: Request):
+    domain = str(request.url).split("/auth/")[0]
+    redirect_uri = f"{domain}/auth/callback/discord"
+    auth_url = f"https://discord.com/api/oauth2/authorize?client_id={settings.auth_discord_client_id}&redirect_uri={redirect_uri}&response_type=code&scope=identify"
+    raise HTTPException(status_code=302, headers={"location": auth_url})
+
+
+@app.get("/auth/callback/discord", response_model=protocol.Token)
+async def callback_discord(
+    code: str,
+    request: Request,
+    db: sqlmodel.Session = Depends(deps.create_session),
+):
+    domain = str(request.url).split("/auth/")[0]
+    redirect_uri = f"{domain}/auth/callback/discord"
+
+    async with aiohttp.ClientSession(raise_for_status=True) as session:
+        # Exchange the auth code for a Discord access token
+        async with session.post(
+            "https://discord.com/api/oauth2/token",
+            data={
+                "client_id": settings.auth_discord_client_id,
+                "client_secret": settings.auth_discord_client_secret,
+                "grant_type": "authorization_code",
+                "code": code,
+                "redirect_uri": redirect_uri,
+                "scope": "identify",
+            },
+        ) as token_response:
+            token_response_json = await token_response.json()
+            access_token = token_response_json["access_token"]
+
+        # Retrieve user's Discord information using access token
+        async with session.get(
+            "https://discord.com/api/users/@me", headers={"Authorization": f"Bearer {access_token}"}
+        ) as user_response:
+            user_response_json = await user_response.json()
+
+    discord_id = user_response_json["id"]
+    discord_username = user_response_json["username"]
+
+    # Get user in our DB linked to the Discord user
+    user: DbUser = get_user_from_provider_id(db, discord_id=discord_id)
```

Comment:
if this can return none, would call it something like `maybe_get_user_from_ ...`, but also just style

Reply:
I have renamed it `query_user_by_provider_id` to match the naming conventions used for this sort of lookup function in the data collection backend repository classes